airflow-core/tests/unit/api_fastapi/core_api/routes/public/test_log.py

```diff
@@ -397,3 +397,60 @@ def test_should_raise_404_when_filtering_on_map_index_for_unmapped_task(self):
         )
         assert response.status_code == 404
         assert response.json()["detail"] == "TaskInstance not found"
+
+    @pytest.mark.parametrize(
+        "supports_external_link, task_id, expected_status, expected_response, mock_external_url",
+        [
+            (
+                True,
+                "task_for_testing_log_endpoint",
+                200,
+                {"url": "https://external-logs.example.com/log/123"},
+                True,
+            ),
+            (
+                False,
+                "task_for_testing_log_endpoint",
+                400,
+                {"detail": "Task log handler does not support external logs."},
+                False,
+            ),
+            (True, "INVALID_TASK", 404, {"detail": "TaskInstance not found"}, False),
+        ],
+        ids=[
+            "external_links_supported_task_exists",
+            "external_links_not_supported",
+            "external_links_supported_task_not_found",
+        ],
+    )
+    def test_get_external_log_url(
+        self, supports_external_link, task_id, expected_status, expected_response, mock_external_url
+    ):
+        with mock.patch(
+            "airflow.utils.log.log_reader.TaskLogReader.supports_external_link",
+            new_callable=mock.PropertyMock,
+            return_value=supports_external_link,
+        ):
+            if mock_external_url:
+                with mock.patch("airflow.utils.log.log_reader.TaskLogReader.log_handler") as mock_log_handler:
+                    mock_log_handler.get_external_log_url.return_value = (
+                        "https://external-logs.example.com/log/123"
+                    )
+
+                    response = self.client.get(
+                        f"/dags/{self.DAG_ID}/dagRuns/{self.RUN_ID}/taskInstances/{task_id}/externalLogUrl/{self.TRY_NUMBER}",
+                        headers={"Accept": "application/json"},
+                    )
+
+                    assert response.status_code == expected_status
+                    assert response.json() == expected_response
+
+                    mock_log_handler.get_external_log_url.assert_called_once() if expected_status == 200 else mock_log_handler.get_external_log_url.assert_not_called()
+            else:
+                response = self.client.get(
+                    f"/dags/{self.DAG_ID}/dagRuns/{self.RUN_ID}/taskInstances/{task_id}/externalLogUrl/{self.TRY_NUMBER}",
+                    headers={"Accept": "application/json"},
+                )
+
+                assert response.status_code == expected_status
+                assert response.json() == expected_response
```

Comment:
Small nit for if / else block.
```suggestion
        response = self.client.get(
            f"/dags/{self.DAG_ID}/dagRuns/{self.RUN_ID}/taskInstances/{task_id}/externalLogUrl/{self.TRY_NUMBER}"
        )

        assert response.status_code == expected_status
        assert response.json() == expected_response

        if mock_external_url and expected_status == 200:
            mock_log_handler.get_external_log_url.assert_called_once()
        elif mock_external_url:
            mock_log_handler.get_external_log_url.assert_not_called()
```